scripting/include/srccoop/entitypatch.inc

```diff
@@ -21,20 +21,58 @@ void SetLocalPlayerEntity(int iEntIndex)
  */
 void SetLocalPlayerEntityEx(CBaseEntity pEntity)
 {
-	#if defined DEBUG
 	if (g_pLocalPlayerEntity != NULL_CBASEENTITY)
 	{
-		LogDebug("WARN: SetLocalPlayerEntity, but one has already been set (%d).", g_pLocalPlayerEntity.entindex);
+		ThrowError("WARN: SetLocalPlayerEntity, but one has already been set (%d).", g_pLocalPlayerEntity.entindex);
 	}
-	#endif
+
 	g_pLocalPlayerEntity = pEntity;
 }
 
 void ClearLocalPlayerEntity()
 {
+	if (g_pLocalPlayerEntity == NULL_CBASEENTITY)
+	{
+		ThrowError("WARN: ClearLocalPlayerEntity, but none was set.");
+	}
+
 	g_pLocalPlayerEntity = NULL_CBASEENTITY;
 }
 
+void SetMaxClientsOverride()
+{
+	if (g_iMaxClientsOverride != -1)
+	{
+		ThrowError("WARN: SetMaxClientsOverride, but one has already been set (%i).", g_iMaxClientsOverride);
+	}
+	
+	CGlobalVars gpGlobals = CPlayerInfoManager().GetGlobalVars();
+	g_iMaxClientsOverride = gpGlobals.GetMaxPlayers();
+	gpGlobals.SetMaxPlayers(1);
+}
+
+void ClearMaxClientsOverride()
+{
+	if (g_iMaxClientsOverride == -1)
+	{
+		ThrowError("WARN: ClearMaxClientsOverride, but none was set.");
+	}
+
+	CGlobalVars gpGlobals = CPlayerInfoManager().GetGlobalVars();
```

Comment:
We should be able to cache gpGlobals as a global variable.

Reply:
Fixed by caching `gpGlobals`.